src/west/app/project.py

```diff
@@ -1243,6 +1243,15 @@ def init_project(self, project):
             # However, west always fetches project data by URL, not name.
             # The user is therefore free to change the URL of this remote.
             project.git(f'remote add -- {project.remote_name} {project.url}')
+
+            # Git submodules that use relative paths resolve them based
+            # on `origin` remote. Adding `origin` allows for usage of
+            # relative submodule paths.
+            try:
+                project.git(f'remote add -- origin {project.url}')
+            except subprocess.CalledProcessError as e:
+                if e.returncode != 128: # Remote already exists
+                    raise(e)
```

Comment:
I'm concerned this may be trusting `git` exit codes a bit too much. I didn't find anything about exit codes in git man pages. This also triggers an exception in the most common case which is counter-intuitive. How about something that does not check `returncode` instead?
```python
# ... based on `origin` remote, make sure we have it
try:
          project.git(f'remote get-url -- origin')
except subprocess.CalledProcessError:
          project.git(f'remote add -- origin {project.url}')
```